Move upgrading/downgrading topics to new guide

adellape opened the pull request:
Preview: http://file.rdu.redhat.com/~adellape/022218/upgradeguide/upgrading/

ahardin-rh commented on upgrading/automated_upgrades.adoc:
```diff
@@ -221,7 +221,7 @@ If you have an older format installation configuration file in
+xref:../install_config/install/quick_install.adoc#defining-an-installation-configuration-file[create one manually].
 
 To start an upgrade with the quick installer:
 
```
include::install_config/upgrading/automated_upgrades.adoc[tag=automated_upgrade_after_reboot]

renders as:

Unresolved directive in <stdin> - include::install_config/upgrading/automated_upgrades.adoc[tag=automated_upgrade_after_reboot]

adellape replied:
Fixt

ahardin-rh commented on upgrading/automated_upgrades.adoc:
```diff
@@ -628,11 +628,11 @@ domain name.
 
 . When you have finished updating your inventory file, follow the instructions in
-xref:../../install_config/cluster_metrics.html#deploying-the-metrics-components[Deploying the Metrics Deployment] to run the *_openshift_metrics.yml_* playbook and
+xref:../install_config/cluster_metrics.html#deploying-the-metrics-components[Deploying the Metrics Deployment] to run the *_openshift_metrics.yml_* playbook and
```
Deploying the Metrics Deployment link is broken (s/.html/.adoc)

ahardin-rh replied:
Verifying the Upgrade is shared between Automated and Manual, but one has an extra step. Is that step missing from the other?

adellape replied:
Fixt

ahardin-rh commented on upgrading/automated_upgrades.adoc:
```diff
@@ -159,13 +159,13 @@ channel and enable the 3.7 channel on each master and node host:
 a playbook to migrate it to external etcd. See
-xref:../../install_config/upgrading/migrating_embedded_etcd.adoc#install-config-upgrading-etcd-data-migration[Migrating Embedded etcd to External etcd] for steps.
+xref:../install_config/upgrading/migrating_embedded_etcd.adoc#install-config-upgrading-etcd-data-migration[Migrating Embedded etcd to External etcd] for steps.
 
```
Both Migrating links are currently broken

adellape replied:
Fixt

ahardin-rh commented on upgrading/index.adoc:
```diff
@@ -0,0 +1,104 @@
+to upgrade a {product-title} cluster. You can use the inventory file that you
+used during inital installation to run the upgrade playbooks. Using this method
+allows you to choose between either
+xref:install-config-upgrading-strategies[upgrade strategy], in-place upgrades or
```
[upgrade strategy]: in-place upgrades or blue green deployments.

(The comma tripped me up at first, as I thought it was a list.)

adellape replied:
Fixt

ahardin-rh commented on upgrading/automated_upgrades.adoc:
```diff
@@ -254,7 +254,7 @@ scenarios:
 
```
In the NOTE box, is Known Issues a reference to the release notes?

adellape replied:
There was previously a Known Issues topic in 3.6. I'll remove this NOTE now.

ahardin-rh commented on upgrading/automated_upgrades.adoc:
```diff
@@ -547,12 +547,12 @@ catalog and service brokers is not supported.
 To upgrade to these features:
```
Otherwise, the next step depends on what additional features you have previously enabled.

adellape replied:
Fixt

ahardin-rh commented on upgrading/manual_upgrades.adoc:
```diff
@@ -416,7 +416,7 @@ updated.
 
 After a cluster upgrade, the default roles
-xref:../../architecture/additional_concepts/authorization.adoc#roles[default
+xref:../architecture/additional_concepts/authorization.adoc#roles[default
```
line 61, should we call RHSM Red Hat Subscription Manager?

adellape replied:
Fixt

ahardin-rh commented on upgrading/manual_upgrades.adoc:
```diff
@@ -510,14 +510,14 @@ ifdef::openshift-enterprise[]
 ====
 You can alternatively use the
-xref:../../install_config/upgrading/blue_green_deployments.adoc#upgrading-blue-green-deployments[blue-green
+xref:../install_config/upgrading/blue_green_deployments.adoc#upgrading-blue-green-deployments[blue-green
```
Link currently broken

adellape replied:
Fixt

ahardin-rh commented on upgrading/manual_upgrades.adoc:
```diff
@@ -890,8 +890,8 @@ provides the example JSON files:
-By default, the xref:../../install_config/install/quick_install.adoc#install-config-install-quick-install[quick] and
-xref:../../install_config/install/advanced_install.adoc#install-config-install-advanced-install[advanced installation]
+By default, the xref:../install_config/install/quick_install.adoc#install-config-install-quick-install[quick] and
+xref:../install_config/install/advanced_install.adoc#install-config-install-advanced-install[advanced installation]
```
In some of the code blocks below, we can probably drop the ==== around the ---- format

adellape replied:
Fixt

ahardin-rh commented on upgrading/blue_green_deployments.adoc:
```diff
@@ -50,7 +50,7 @@ image::blue-green-deployment.gif[Blue-Green Deployment Upgrade]
 
 After you have upgraded your master and etcd hosts using method described for
-xref:../../install_config/upgrading/manual_upgrades.adoc#upgrading-masters[In-place
+xref:../install_config/upgrading/manual_upgrades.adoc#upgrading-masters[In-place
```
this link is currently broken

adellape replied:
Fixt

ahardin-rh commented on upgrading/blue_green_deployments.adoc:
```diff
@@ -260,7 +260,7 @@ new image streams.
 introduce new image streams for builder images for users of Source-to-Image
 (S2I). Upon import, any builds or deployments configured with
-xref:../../dev_guide/builds/triggering_builds.adoc#image-change-triggers[image
+xref:../dev_guide/builds/triggering_builds.adoc#image-change-triggers[image
```
what does Warming mean?

adellape replied:
Added a line later in the section: The green
nodes are then considered _warmed_, and everything else can be moved over using
node evacuation in a later step, proceeding more quickly as a result.

ahardin-rh commented on upgrading/blue_green_deployments.adoc:
```diff
@@ -286,10 +286,10 @@ $ oc adm manage-node --schedulable=true --selector=color=green
 
 . Update the default image streams and templates as described in
-xref:../../install_config/upgrading/manual_upgrades.adoc#updating-the-default-image-streams-and-templates[Manual In-place Upgrades].
+xref:../install_config/upgrading/manual_upgrades.adoc#updating-the-default-image-streams-and-templates[Manual In-place Upgrades].
```
Both links are currently broken

adellape replied:
Fixt

ahardin-rh commented on upgrading/migrating_embedded_etcd.adoc:
```diff
@@ -19,7 +19,7 @@ etcd API version since {product-title} 3.6 defaults to v3. Also, since
 deployments with embedded etcd with the etcd API version v2 need to migrate to
 the external etcd first,
-xref:../../install_config/upgrading/migrating_etcd.adoc#install-config-upgrading-etcd-data-migration[followed
+xref:../install_config/upgrading/migrating_etcd.adoc#install-config-upgrading-etcd-data-migration[followed
```
link is currently broken

adellape replied:
Fixt

ahardin-rh commented on upgrading/migrating_etcd.adoc:
```diff
@@ -27,7 +27,7 @@ performed using openshift-ansible version 3.6.173.0.21 or later.
 Until {product-title} 3.6, it was possible to deploy a cluster with an embedded
 etcd. As of {product-title} 3.7, this is no longer possible. See
-xref:../../install_config/upgrading/migrating_embedded_etcd.adoc#install-config-upgrading-ee-etcd-data-migration[Migrating
+xref:../install_config/upgrading/migrating_embedded_etcd.adoc#install-config-upgrading-ee-etcd-data-migration[Migrating
```
this link is currently broken

adellape replied:
Fixt

ahardin-rh commented on upgrading/migrating_etcd.adoc:
```diff
@@ -58,7 +58,7 @@ function as expected.
 migrate to the external etcd before migrating data. See
-xref:../../install_config/upgrading/migrating_embedded_etcd.adoc#install-config-upgrading-ee-etcd-data-migration[Migrating
+xref:../install_config/upgrading/migrating_embedded_etcd.adoc#install-config-upgrading-ee-etcd-data-migration[Migrating
 Embedded etcd to External etcd].
```
s/RHSM/Red Hat Subscription Manager  🤔

adellape replied:
Fixt

ahardin-rh commented on upgrading/migrating_etcd.adoc:
```diff
@@ -103,7 +103,7 @@ The following procedure describes the steps required to successfully migrate the
 
 . +++<b>Create an etcd backup.</b>+++ See
-xref:../../admin_guide/backup_restore.adoc#cluster-backup[Backup and Restore]
+xref:../admin_guide/backup_restore.adoc#cluster-backup[Backup and Restore]
```
When you click thru, the Backup and Restore topic is currently missing a lot of content in the build. Not sure what's up.

adellape replied:
Hmm, looks like a problem in master.

adellape commented on upgrading/manual_upgrades.adoc:
```diff
@@ -1347,7 +1347,7 @@ ifdef::openshift-enterprise[]
 for {product-title} 3.7.
 
-See the xref:../../release_notes/ocp_3_7_release_notes.adoc#release-notes-ocp-3-7-release-notes[{product-title} 3.7 Release Notes] to review the latest release notes.
+See the xref:../release_notes/ocp_3_7_release_notes.adoc#release-notes-ocp-3-7-release-notes[{product-title} 3.7 Release Notes] to review the latest release notes.
```
Should be fixt

adellape commented:
@ahardin-rh Addressed your comments and moved the Downgrading topic to the new guide, per discussion. New preview:

 http://file.rdu.redhat.com/~adellape/022218/upgradeguide/upgrading/

adellape commented:
@vikram-redhat This is ready to merge, but it's going to need redirects put in place for the docs.openshift sites. See topic_map for changes. Not sure if I should merge this now or not, please LMK. Also needs the needful done on the Portal side.

ahardin-rh commented on upgrading/automated_upgrades.adoc:
```diff
@@ -254,7 +248,7 @@ scenarios:
 - Upgrading existing {product-title} 3.6 clusters to 3.7
 - Upgrading existing {product-title} 3.7 clusters to the latest
-xref:../../release_notes/ocp_3_7_release_notes.html#ocp-37-asynchronous-errata-updates[asynchronous
+xref:../release_notes/ocp_3_7_release_notes.html#ocp-37-asynchronous-errata-updates[asynchronous
```
This asynchronous errata updates link is broken in the preview build

adellape replied:
Fixt

vikram-redhat commented:
@adellape - sure. Can you add the redirects that are required to the Redirects Register? https://docs.google.com/spreadsheets/d/1KH2_p54Fr2KRFTUGzQdpqExox7iL9RM4_yiPNQpArjk/edit#gid=0.

Also, this mostly affects the install config guide, right? So won't require the dedicated and online labels.

adellape commented:
> Also, this mostly affects the install config guide, right? So won't require the dedicated and online labels.

@vikram-redhat Yeah, I can drop them now thx to new workflow!

adellape commented:
> Can you add the redirects that are required to the Redirects Register?

@vikram-redhat Done.

ahardin-rh commented on upgrading/manual_upgrades.adoc:
```diff
@@ -805,11 +805,11 @@ node2.example.com           Ready                      165d
 the router image. To upgrade your router without disrupting services, you must
 have previously deployed a
-xref:../../admin_guide/high_availability.adoc#configuring-a-highly-available-service[highly-available
+xref:../admin_guide/high_availability.adoc#configuring-a-highly-available-service[highly-available
```
Hmm I don't think highly-available should be hyphenated, but it is in the admin topic too

adellape replied:
Oops you said that and I missed updating it, sry.

ahardin-rh commented on upgrading/os_upgrades.adoc:
```diff
@@ -35,7 +35,7 @@ that all of the rules in other services are correct.
 affected, or preserve the `iptables` state. Both processes are described in the
-xref:../../admin_guide/iptables.adoc#admin-guide-iptables[{product-title}
+xref:../admin_guide/iptables.adoc#admin-guide-iptables[{product-title}
 IPtables] topic. The `ovs` flow rules do not need to be saved, but restarting
```
total nit, but this should be iptables instead of IPtables to be consistent with the style of the page it is pointing to

vikram-redhat commented:
Thanks. @adellape - ping me when you are ready to merge this. I will set the redirects up and then give you the go ahead to merge. Since this is a hot topic, I want to make sure we are synchronized on this.

ahardin-rh commented on _topic_map.yml:
```diff
@@ -510,6 +489,28 @@ Topics:
+  File: os_upgrades
+- Name: Migrating Embedded etcd to External etcd
+  File: migrating_embedded_etcd
+- Name: Migrating etcd Data (v2 to v3)
```
Checking if this v2 to v3 topic should stick around or be retired

adellape replied:
This is about etcd v2 to v3 and not OpenShift overall, so it should stay in for 3.7's docs (probably removed for 3.9's).

adellape commented:
@ahardin-rh Thx, comments addressed!

@vikram-redhat This is ready to merge. If you do the redirects and end up wanting to merge (guess it needs cherry-picking at that point too), feel free. Or I'll try to catch you Monday since it's already into Friday there now.

vikram-redhat commented:
@adellape - haven't finished this. Hold off on merging/cherry picking. I will work with Drupal team to synchronize it.

vikram-redhat commented:
Merging this and then CP to 3.7 and 3.9 branches. Redirects for docs.openshift.com/org setup using this: https://github.com/openshift/openshift-docs/pull/7954.

Portal rules setup but can't proceed till this is merged and CPed so merging this.

vikram-redhat commented:
/cherrypick enterprise-3.7

vikram-redhat commented:
@adellape - ugh looks like conflicts. Will try and do a rebase locally before going ahead.

vikram-redhat commented:
@adellape - CP'ed to 3.7 via: https://github.com/openshift/openshift-docs/pull/7955 and added some minor link fixes.

vikram-redhat commented:
@adellape @ahardin-rh @aheslin 

All redirects are in place for both d.o.c (commercial and origin) and portal and the books are live.

d.o.c: https://docs.openshift.com/container-platform/3.7/upgrading/index.html
Portal: https://access.redhat.com/documentation/en-us/openshift_container_platform/3.7/html-single/upgrading_clusters/

Next: CPing to 3.9 branch. Hopefully, will be easier now that 3.7 conflicts are resolved.

(If you see any issues with redirects please let me know. My redirect fu isn't that great.)

vikram-redhat commented:
And 3.9 is done as well. https://github.com/openshift/openshift-docs/pull/7956

adellape commented:
@vikram-redhat Thanks so much!
